### ISSUE TITLE
Wrong columns in query result in query inspector - table panel doesn't work - joins not working properly

### ISSUE THREAD

**Issue by MaestroGlanz** [grafana/grafana#37179]
I'm using Grafana while being connected to the gnucash-database.

I try to use a table panel to replicate the ui of gnucash (partially at least)

In the table panel I defined the following (test) query:
```
select distinct s1.value_num as wert, s1.value_denom as umrechnung, t1.post_date as time, t1.description as subject
from splits s1
inner join splits s2 on s1.tx_guid=s2.tx_guid
inner join transactions t1 on t1.guid=s2.tx_guid
order by time ASC limit 4;
```

In the query inspector, the result is:
```
results:Object
D:Object
  frames:Array[1]
    0:Object
      schema:Object
        name:"results"
        meta:Object
        fields:Array[5]
          0:Object
            name:"Time"
            type:"time"
            typeInfo:Object
          1:Object
            name:"umrechnung"
            type:"number"
            typeInfo:Object
            labels:Object
          2:Object
            name:"umrechnung"
            type:"number"
            typeInfo:Object
            labels:Object
          3:Object
            name:"wert"
            type:"number" 
            typeInfo:Object
            labels:Object
          4:Object
            name:"wert"
            type:"number"
            typeInfo:Object
            labels:Object
      data:Object
        values:Array[5]
          0:Array[1273834740000,1292324340000,1293706740000]
          1:Array[3]
            0:null
            1:null
            2:100
          2:Array[100,100,null]
          3:Array[3]
            0:null
            1:null
            2:260
          4:Array[0,214353,null]
```

The result, which is shown in the panel is even more strange, as it contains query data in the column names:

Time umrechnung, ABSCHLUSS PER 31.12.2010 SALDO RECHNUNGSABSCHLUSS, umrechnung Er�ffnungssaldo, wert ABSCHLUSS PER 31.12.2010 SALDO RECHNUNGSABSCHLUSS, wert Er�ffnungssaldo

The query from the beginning is a shortened version of a query, in which I got the following error (I can add it on demand):
```
convert value to float failed: metricIndex 4 type can't be converted to float
```

- Grafana version: 8.0.6
- Data source type & version: Postgres 12.7
- OS Grafana is installed on: Ubuntu Regolith 20.10
- User OS & Browser: As before, Firefox 90
- Grafana plugins: old pie chart

**Comment by MaestroGlanz** [comment]
That's what it should look like (same query in psql terminal):
```
  wert      | umrechnung |        time                     |                       subject                       
------------+-----------------+-----------------------------+-----------------------------------------------------
           0  |               100 | 2010-05-14 10:59:00 | Er�ffnungssaldo
 -214353 |               100 | 2010-12-14 10:59:00 | Er�ffnungssaldo
  214353 |               100 | 2010-12-14 10:59:00 | Er�ffnungssaldo
        260 |               100 | 2010-12-30 10:59:00 | ABSCHLUSS PER 31.12.2010   SALDO RECHNUNGSABSCHLUSS

```

**Comment by zuchka** [comment]
Hi @MaestroGlanz 

Based on the response to your comment on similar/duplicate issue #37151, it looks like #37195 should fix this. You could test that now by running Grafana on `main`, or after version 8.1 releases in a few weeks.

**Comment by zuchka** [comment]
Hi @MaestroGlanz 

Thanks for making this issue. I see that you also commented on issue #37151, and there they mentioned #37195. 

Can you please report the specific datasource plugin that you are using? I just realized that the other issue your commented on was about InfluxDB but here you appear to be using the [built-in Postgres Datasource plugin](https://grafana.com/grafana/plugins/postgres/)? Is that correct?

**Comment by MaestroGlanz** [comment]
@zuchka That is correct. Internal postgres data source and stock table panel. Pretty basic.

**Comment by MaestroGlanz** [comment]
@marefr I want this as time series. The table contains a time stamp. What I do in psql with "WHERE post_date > $startdate AND post_date < $enddate" should be handled implicitly by grafana. As bigger and smaller works in psql, I can conclude, that the format complies to sql standards.

**Comment by marefr** [comment]
> I want this as time series. The table contains a time stamp

Depends on how you want to visualize it, if just tabular format (a table) you can have a column for time, right? See my example:
![Selection_1041](https://user-images.githubusercontent.com/1668778/131574596-04a065bb-7b69-41e8-9fed-341590d25b4d.png)

If you still want to use time series format please explain what you're trying to achieve.

> What I do in psql with "WHERE post_date > $startdate AND post_date < $enddate" should be handled implicitly by grafana. As bigger and smaller works in psql, I can conclude, that the format complies to sql standards.

This feels unrelated to your issue and more of a question, but please refer to the [macros documentation](https://grafana.com/docs/grafana/latest/datasources/postgres/#macros), for example using `$__timeFrom()`, `$__timeTo()` or `$__timeFilter(dateColumn)`.
